python/ray/__init__.py

```diff
@@ -37,7 +37,7 @@
 if os.path.exists(so_path):
     import ctypes
     from ctypes import CDLL
-    CDLL(so_path, ctypes.RTLD_GLOBAL)
+    CDLL(so_path, ctypes.RTLD_LOCAL)
```

Comment:
Should probably leave a comment here pointing to the issue this fixed

Reply:
Definitely will add!

Reply:
@ijrsvt Can 有 give an example which symbols conflict. We used linker script `ray_exported_symbols.lds` and `ray_version_script.lds` to limit exported symbols. And using `RTLD_GLOBAL` by purpose so that _streaming.so can using symbols in _raylet.so

Reply:
The conflicting symbols were in the GRPC library. The problem only showed up on Linux, and when `pyarrow` was imported before `ray`.  The specific symbol that was segfaulting was 
`google::protobuf::internal::AssignDescriptors`.

Reply:
Can you give more complete detailed log for this symbol segfault. I tink we didn't expose protobuf symbols in `ray_exported_symbols.lds/ray_version_script.lds`. And this import order issue exists before `RTLD_GLOBAL`.

Reply:
It is [here]("https://gist.github.com/ijrsvt/f974ca883b2c191c438243df7d5eda3b"). It appears to be some issue where there is a conflicting symbol from gRPC.

Reply:
@chaokunyang Can you look more into this and why the symbols are conflicting despite the version linker script? This is blocking the current release I believe.

Reply:
@pcmoritz I'm looking into it. There may be some symbols exported unexpectedly despite of the version linker script, like template. If that's the reason, maybe we can use `__attribute__((visibility(....)))`. If not, we'll need more time for this.